components/blitz/src/ome/services/roi/GeomTool.java

```diff
@@ -354,6 +358,143 @@ public void handle(int x, int y) {
 
     }
 
+    public ShapeStats [] getStatsRestricted(
+            List<Long> shapeIds, 
+            int zForUnattached, int tForUnattached,
+            int[] channels) {
+        if (shapeIds == null || shapeIds.isEmpty()) 
+            throw new ApiUsageException("Provide a non empty list of shape ids.");
+
+        final Session session = factory.getSession();
+        final List<ShapeStats> shapeStats = 
+            new ArrayList<ShapeStats>(shapeIds.size());
+        // we lump together shapes that belong to the same z/t
+        // in order to not have to read the same z/t twice
+        final Map<String, List<ome.model.roi.Shape>> zt_lookup =
```

Comment:
guava multimap could be used in this case

Reply:
Is this a dependency already? Otherwise I'd ask myself what we are gaining given. Performance wise nothing I's suspect since the number of entries are low for a hashmap anyhow.

Reply:
we already have a guava dependency but this is not a blocker just a suggestion